### PR TITLE
Google maps script fix

### PR DESCRIPTION
**Pull request by Gnito**
- Handle feature_enabled?(:location_search) checks a little bit better on index.haml
- Add scripts only on page level (google.maps should not be handled on partials)

### CODE REVIEW

File: app/views/homepage/index.haml [hunk 1]
```diff
@@ -124,7 +128,7 @@
           - if @view_type.eql?("grid")
             %div{class: @category_menu_enabled ? "home-fluid-thumbnail-grid-narrow" : "home-fluid-thumbnail-grid-wide"}
               .home-fluid-thumbnail-grid
-                = render :partial => "grid_item", :collection => @listings, :as => :listing, locals:{show_distance: show_distance}
+                = render :partial => "grid_item", :collection => @listings, :as => :listing, locals:{show_distance: location_search_in_use}
```
Comment (Gnito) [inline app/views/homepage/index.haml]: "locals:{" is changed to "locals: {" in another branch which probably goes to master first (so, some conflicts ahead)

File: app/views/homepage/_location_bar.haml [hunk 2]
```diff
@@ -1,7 +1,3 @@
--content_for(:location_search) do
-  - if(feature_enabled?(:location_search) && search_engine == :zappy)
-    = javascript_include_tag "https://maps.google.com/maps/api/js?sensor=true&libraries=places"
```
Comment (rap1ds) [inline app/views/homepage/_location_bar.haml]: This is good :+1: Definitely we shouldn't require the map component ad hoc in partials :+1:

File: app/controllers/homepage_controller.rb [hunk 3]
```diff
@@ -51,21 +51,23 @@ def index
         raise ArgumentError.new("Unknown view_type #{@view_type}")
       end
 
-    search_result = find_listings(params, per_page, compact_filter_params, includes.to_set)
+    main_search = (feature_enabled?(:location_search) && search_engine == :zappy) ? MarketplaceService::API::Api.configurations.get(community_id: @current_community.id).data[:main_search] : :keyword
+    location_search_in_use = (feature_enabled?(:location_search) && search_engine == :zappy && main_search == :location)
```
Comment (rap1ds) [inline app/controllers/homepage_controller.rb]: These two lines seem to have a lot of repetition and the logic feels a bit too complicated. Can we figure out a way to make it simpler?

File: app/controllers/homepage_controller.rb [hunk 4]
```diff
@@ -243,8 +247,8 @@ def shapes
     ListingService::API::Api.shapes
   end
 
-  def location_search_params(latlng, search_engine_in_use)
-      if(feature_enabled?(:location_search) && latlng.present? && search_engine_in_use == :zappy)
+  def location_search_params(latlng, search_engine_in_use, main_search)
+      if(feature_enabled?(:location_search) && latlng.present? && search_engine_in_use == :zappy && main_search == :location)
         coordinates = latlng.split(',')
```
Comment (rap1ds) [inline app/controllers/homepage_controller.rb]: Ruby trick:

```
lat, lng = latlng.split(',')
```

File: app/controllers/homepage_controller.rb [hunk 5]
```diff
@@ -243,8 +247,8 @@ def shapes
     ListingService::API::Api.shapes
   end
 
-  def location_search_params(latlng, search_engine_in_use)
-      if(feature_enabled?(:location_search) && latlng.present? && search_engine_in_use == :zappy)
+  def location_search_params(latlng, search_engine_in_use, main_search)
+      if(feature_enabled?(:location_search) && latlng.present? && search_engine_in_use == :zappy && main_search == :location)
```
Comment (rap1ds) [inline app/controllers/homepage_controller.rb]: Maybe we can pass all search parameters to the adapter and let the search engine to decide which once to use?

File: app/controllers/homepage_controller.rb [hunk 6]
```diff
@@ -51,21 +51,21 @@ def index
         raise ArgumentError.new("Unknown view_type #{@view_type}")
       end
 
+    main_search = location_search_available ? MarketplaceService::API::Api.configurations.get(community_id: @current_community.id).data[:main_search] : :keyword
+    location_search_in_use = location_search_available && main_search == :location
```
Comment (rap1ds) [inline app/controllers/homepage_controller.rb]: The `location_search_available` is redundant in this line. The previous line takes that already into account.

**Comment by rap1ds** [comment]
Looks good! Only on minor comment

File: app/controllers/homepage_controller.rb (hunk 6 above)
Comment (Gnito) [inline app/controllers/homepage_controller.rb]: true :)

**Pull request merged**